src/formatDate.js

```diff
@@ -8,7 +8,58 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(separator);
+
+  const dateCompare = {};
+
+  fromFormat.forEach((part, index) => {
+    switch (part) {
+      case 'YYYY':
+        dateCompare['YYYY'] = dateParts[index];
+        break;
+      case 'YY':
+        dateCompare['YY'] = dateParts[index];
+        break;
+      case 'MM':
+        dateCompare['MM'] = dateParts[index];
+        break;
+      case 'DD':
+        dateCompare['DD'] = dateParts[index];
+        break;
+      default:
+        break;
+    }
+  });
+
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const newDateParts = toFormat.slice(0, -1).map((part) => {
+    switch (part) {
+      case 'YYYY':
+        if (dateCompare['YY']) {
+          const yy = parseInt(dateCompare['YY']);
+
+          return yy < 30 ? `20${dateCompare['YY']}` : `19${dateCompare['YY']}`;
```

Comment:
The logic for converting a two-digit year to a four-digit year is correct. It properly checks if the year is less than 30 and prefixes it with '20', otherwise with '19'. This is in line with the task requirements.